plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -2455,17 +2455,23 @@ public ConnectorTableProperties getTableProperties(ConnectorSession session, Con
         HiveTableHandle hiveTable = (HiveTableHandle) table;
 
         List<ColumnHandle> partitionColumns = ImmutableList.copyOf(hiveTable.getPartitionColumns());
-        List<HivePartition> partitions = partitionManager.getOrLoadPartitions(metastore, new HiveIdentity(session), hiveTable);
-
-        TupleDomain<ColumnHandle> predicate = createPredicate(partitionColumns, partitions);
-
+        TupleDomain<ColumnHandle> predicate = hiveTable.getEnforcedConstraint();
         Optional<DiscretePredicates> discretePredicates = Optional.empty();
-        if (!partitionColumns.isEmpty()) {
-            // Do not create tuple domains for every partition at the same time!
-            // There can be a huge number of partitions so use an iterable so
-            // all domains do not need to be in memory at the same time.
-            Iterable<TupleDomain<ColumnHandle>> partitionDomains = Iterables.transform(partitions, (hivePartition) -> TupleDomain.fromFixedValues(hivePartition.getKeys()));
-            discretePredicates = Optional.of(new DiscretePredicates(partitionColumns, partitionDomains));
+
+        if (hiveTable.getPartitionNames().isEmpty()) {
```

Comment:
Does empty `partition names` assert that `applyFilter` was applied? Comment would be nice.

Reply:
If the partitionNames is empty then it
 * means the `applyFilter` is applied and number of partitions is less than the threshold
 * `applyFilter` is not yet applied.

Reply:
> means the applyFilter is applied and number of partitions is less than the threshold

You mean greater?